File: relationship-manager/auth-service/src/main/java/com/bank/rm/auth/repository/UserRepository.java
package com.bank.rm.auth.repository;

import com.bank.rm.auth.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.Optional;
import java.util.UUID;

public interface UserRepository extends JpaRepository<User, UUID> {
    Optional<User> findByEmail(String email);
    Optional<User> findByAnonymousSessionId(UUID anonymousSessionId);
    boolean existsByEmail(String email);
}

File: relationship-manager/auth-service/src/main/kotlin/com/bank/rm/auth/repository/UserRepository.kt
package com.bank.rm.auth.repository

import com.bank.rm.auth.domain.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface UserRepository : JpaRepository<User, UUID> {
    fun findByEmail(email: String): User?
    fun findByPhone(phone: String): User?
    fun existsByEmail(email: String): Boolean
}
